src/extension/features/budget/fund-half/index.ts

```diff
@@ -0,0 +1,160 @@
+        "button div:contains('Underfunded')"
+      );
+      const underfundedButton = $(underfundedButtonText).parent();
+
+      this.lowHalf = this._getSelectedCategoriesFundAmounts(selectedBudgetCategories, HalfType.Low);
+      this.highHalf = this._getSelectedCategoriesFundAmounts(
+        selectedBudgetCategories,
+        HalfType.High
+      );
+
+      let lowHalfSum: number = 0;
+      this.lowHalf.forEach((a) => (lowHalfSum += a.amount));
+
+      let highHalfSum: number = 0;
+      this.highHalf.forEach((a) => (highHalfSum += a.amount));
+
+      if (lowHalfSum === highHalfSum) {
+        // We only need one button.  This may be a case where all selected budget category's Target amounts were even numbers.
+        // Or the case where there are multiple odd amounts that equalized the amounts.
+        $(underfundedButton)
+          .parent()
+          .append(this.halfFundButton('Fund Half', lowHalfSum, HalfType.Exact));
+      } else {
+        $(underfundedButton)
+          .parent()
+          .append(this.halfFundButton('Fund Low Half', lowHalfSum, HalfType.Low));
+
+        $(underfundedButton)
+          .parent()
+          .append(this.halfFundButton('Fund High Half', highHalfSum, HalfType.High));
+      }
+    }
+
+    $('button.budget-inspector-button.fund-half').on('mousedown', (event) => {
+      const halfTypeString = event.delegateTarget.attributes.getNamedItem('data-half-type')!.value;
+
+      // This will treat HalfType Exact like Low.
+      const selectedBudgetCategories =
+        halfTypeString.valueOf() !== HalfType.High ? this.lowHalf : this.highHalf;
+
+      selectedBudgetCategories.forEach((selectedBudgetCategory) => {
+        this._fundAmountToBudgetCategory(
+          selectedBudgetCategory.budgetCategoryId,
+          selectedBudgetCategory.amount
+        );
+      });
+    });
+  }
+
+  _getSelectedCategoriesFundAmounts(
+    selectedBudgetCategories: YNABBudgetMonthDisplayItem[],
+    halfType: HalfType
+  ): BudgetCategoryAmount[] {
+    const selectedBudgetCategoryAmounts: BudgetCategoryAmount[] = [];
+    selectedBudgetCategories.forEach((selectedBudgetCategory) => {
+      const budgeCategoryAmount: BudgetCategoryAmount = {
+        budgetCategoryId: selectedBudgetCategory.categoryId,
+        amount: this._getHalfAmount(selectedBudgetCategory.goalTarget, halfType),
+      };
+
+      selectedBudgetCategoryAmounts.push(budgeCategoryAmount);
+    });
+
+    return selectedBudgetCategoryAmounts;
+  }
+
+  _getSelectedBudgetCategories = () => {
+    const budgetService = getBudgetService();
+    if (
+      budgetService !== undefined &&
+      budgetService != null &&
+      budgetService.checkedRowsCount > 0
+    ) {
+      return budgetService.checkedRows;
+    }
+
+    return null;
+  };
+
+  _getHalfAmount = (amount: number, halfType: HalfType) => {
+    amount = amount / 10;
+    if (amount & 1) {
```

Comment:
This is a bit unconventional way (at least in JavaScript) to check if something is odd, I think `amount % 2 === 1` will be easier to understand for whoever will work with this code in the future

Reply:
Thanks, fixed!